lib/pagy/extras/headers.rb

```diff
@@ -0,0 +1,35 @@
+# See the Pagy documentation: https://ddnexus.github.io/pagy/extras/headers
+# encoding: utf-8
+# frozen_string_literal: true
+
+class Pagy
+  # Add specialized backend methods to add pagination response headers
+  module Backend ; private
+
+    include Helpers
+
+    def pagy_headers_merge(pagy)
+      response.headers.merge!(pagy_headers(pagy))
+    end
+
+    def pagy_headers(pagy)
+      hash = pagy_headers_hash(pagy)
+      { 'Links'    => hash[:links].map{|rel, link| %(<#{link}>; rel="#{rel}")}.join(', ') }.tap do |h|
+        h['Items'] = h['Per-Page'] = hash[:items]
+        h['Count'] = h['Total']    = hash[:count] if hash.key?(:count)
+      end
+    end
+
+    def pagy_headers_hash(pagy)
+      countless = defined?(Pagy::Countless) && pagy.is_a?(Pagy::Countless)
+      rels      = { first: 1, prev: pagy.prev, next: pagy.next }.tap{ |r| r[:last] = pagy.last unless countless }
+      a, b      = pagy_url_for(Frontend::MARKER, pagy, :url).split(Frontend::MARKER, 2)
+      links     = Hash[rels.map{|rel, n|[rel, "#{a}#{n}#{b}"] if n}.compact]
```

Comment:
if none of them is nil this will be nil ?

```
[].compact!
=> nil
```

Reply:
not sure to understand "If none of them": did you mean "if all them are nil"?
BTW, there will be always at least the first page link (even if it is empty), so that should not happen... however I will try to avoid asumptions. Thanks

Reply:
if no `nil` is removed the result will be nil

Reply:
I still don't get your comment at all :/ (maybe my morning-dumbness... or just dumbness)
How `[].compact!` relates to the code there? Is that a suggestion or a warning?